tests: driver: pwm: pwm test build failure

### Describe the bug

on mr_canhubk3/s32k344 build with

git bisect found below commit introduce issue

commit 0bdb9595b1a394fb8295998ebb4f67edb70b5d7f
```
Author: Emilio Benavente <emilio.benavente@nxp.com>
Date:   Tue Jul 8 14:51:09 2025 -0500

    drivers: misc: mcux_flexio: Set the DOZEN bit
    
    Setting the DOZEN bit in the flexio so the soc
    does not force the peripheral to go into
    low power mode when the soc is in idle.
    
    Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

 drivers/misc/mcux_flexio/mcux_flexio.c | 2 ++
 1 file changed, 2 insertions(+)
```

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b mr_canhubk3/s32k344 tests/drivers/pwm
```

### Relevant log output

```shell
[86/222] Linking C static library zephyr/soc/soc/s32k344/s32k3/libsoc__nxp__s32__s32k3.a[87/222] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj[88/222] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_error_hook.c.obj[89/222] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_rules.c.obj[90/222] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_defaults.c.obj[91/222] Linking C static library zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a[92/222] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_eirq_nxp_s32.c.obj[93/222] Linking C static library zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a[94/222] Building C object zephyr/drivers/misc/nxp_s32_emios/CMakeFiles/drivers__misc__nxp_s32_emios.dir/nxp_s32_emios.c.obj[95/222] Linking C static library zephyr/drivers/misc/nxp_s32_emios/libdrivers__misc__nxp_s32_emios.a[96/222] Building C object zephyr/drivers/misc/mcux_flexio/CMakeFiles/drivers__misc__mcux_flexio.dir/mcux_flexio.c.objFAILED: zephyr/drivers/misc/mcux_flexio/CMakeFiles/drivers__misc__mcux_flexio.dir/mcux_flexio.c.obj nter-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/tests/drivers/pwm/pwm_loopback=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/misc/mcux_flexio/CMakeFiles/drivers__misc__mcux_flexio.dir/mcux_flexio.c.obj -MF zephyr/drivers/misc/mcux_flexio/CMakeFiles/drivers__misc__mcux_flexio.dir/mcux_flexio.c.obj.d -o zephyr/drivers/misc/mcux_flexio/CMakeFiles/drivers__misc__mcux_flexio.dir/mcux_flexio.c.obj -c /home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/misc/mcux_flexio/mcux_flexio.c/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/misc/mcux_flexio/mcux_flexio.c: In function 'mcux_flexio_init':/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/misc/mcux_flexio/mcux_flexio.c:135:23: error: 'flexio_config_t' {aka 'struct _flexio_config_'} has no member named 'enableInDoze'; did you mean 'enableInDebug'? 135 | flexio_config.enableInDoze = true; | ^~~~~~~~~~~~ | enableInDebugninja: build stopped: subcommand failed.
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: All
- toolchain: 0.17.1
- commit: 7a8bd3dbe7528ff591a217b0dd81663c34a3c370

### Additional Context

_No response_

--- a/drivers/misc/mcux_flexio/mcux_flexio.c
+++ b/drivers/misc/mcux_flexio/mcux_flexio.c
@@ -132,7 +132,10 @@ static int mcux_flexio_init(const struct device *dev)
 	k_mutex_init(&data->lock);
 
 	FLEXIO_GetDefaultConfig(&flexio_config);
+#if !(defined(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT) && \
+		(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT == 0))
 	flexio_config.enableInDoze = true;
+#endif
 
 	FLEXIO_Init(config->base, &flexio_config);
 	config->irq_config_func(dev);

--- a/drivers/spi/spi_mcux_flexio.c
+++ b/drivers/spi/spi_mcux_flexio.c
@@ -136,14 +136,20 @@ static void spi_flexio_master_init(FLEXIO_SPI_Type *base, flexio_spi_master_conf
 
 	/* Configure FLEXIO SPI Master */
 	ctrlReg = base->flexioBase->CTRL;
-	ctrlReg &= ~(FLEXIO_CTRL_DOZEN_MASK | FLEXIO_CTRL_DBGE_MASK |
-			FLEXIO_CTRL_FASTACC_MASK | FLEXIO_CTRL_FLEXEN_MASK);
+	ctrlReg &= ~(FLEXIO_CTRL_DBGE_MASK | FLEXIO_CTRL_FASTACC_MASK | FLEXIO_CTRL_FLEXEN_MASK);
+#if !(defined(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT) && \
+		(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT == 0))
+	ctrlReg &= ~FLEXIO_CTRL_DOZEN_MASK;
+#endif
 	ctrlReg |= (FLEXIO_CTRL_DBGE(masterConfig->enableInDebug) |
 		FLEXIO_CTRL_FASTACC(masterConfig->enableFastAccess) |
 		FLEXIO_CTRL_FLEXEN(masterConfig->enableMaster));
+#if !(defined(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT) && \
+		(FSL_FEATURE_FLEXIO_HAS_DOZE_MODE_SUPPORT == 0))
 	if (!masterConfig->enableInDoze) {
 		ctrlReg |= FLEXIO_CTRL_DOZEN_MASK;
 	}
+#endif
 
 	base->flexioBase->CTRL = ctrlReg;
 

